kernel/enter.c

```diff
@@ -446,15 +470,26 @@ int myst_enter_kernel(myst_kernel_args_t* args)
     }
     else
     {
-#ifdef MYST_ENABLE_EXT2FS
+        bool loaded_fs = false;
 
+#if defined(MYST_ENABLE_EXT2FS)
         /* setup and mount the EXT2 file system */
-        if (_setup_ext2(args->rootfs) != 0)
-            ERAISE(-EINVAL);
+        if (_setup_ext2(args->rootfs) == 0)
+        {
+            loaded_fs = true;
+        }
+#endif
 
-#else  /* MYST_ENABLE_EXT2FS */
-        myst_panic("ext2 not supported");
-#endif /* MYST_ENABLE_EXT2FS */
+#if defined(MYST_ENABLE_HOSTFS) && defined(MYST_DEBUG)
```

Comment:
There should be a comment somewhere says: `if you enable HOSTFS, additionally if you are in debug mode, then we allow the hostfs to be mounted as root fs`

Reply:
We should use the `debug` flag in config.json. That's a more truthful reflection of whether the enclave is in debug mode or not.

Reply:
I will add to the file systems doc when read.

Good idea about the debug flag. Rather than relying on the config.json flag, I added a function to use **oe_get_report_v2()** and **oe_parse_report()** and then I check that then I check the following mask.

```
parsed_report.identity.attributes & OE_REPORT_ATTRIBUTES_DEBUG
``